#include "arkmidnight.h"
#include "ui_arkmidnight.h"
#include "guide1.h"
#include "game.h"
#include <QDebug>

int ArkMidnight::state[10] = {0};

ArkMidnight::ArkMidnight(QWidget *parent, int _n)
    : QMainWindow(parent)
    , ui(new Ui::ArkMidnight)
{
    ui->setupUi(this);
    this->setWindowIcon(QIcon(":/setting/image/icon.png"));
    this->setWindowTitle("ArkMidnight");
    ui->stackedWidget->setCurrentIndex(_n);
}

ArkMidnight::~ArkMidnight()
{
    delete ui;
}

void ArkMidnight::vis(){
    this->setVisible(true);
}

void ArkMidnight::vis2(){
    this->setVisible(true);
    ui->stackedWidget->setCurrentIndex(1);
    QImage back_img;
    if (state[1] == 1){
        back_img.load(":/setting/image/bg_win.png");
        ui->label1->setText(QString::asprintf("Stage %d  ÊåëÊàòÊàêÂäü!", state[0]));
        ui->label2->setText(QString::asprintf("‰Ω†ÂÖ±ÂáªÊùÄ‰∫Ü%dÂè™ÊÄ™\nÂâ©‰ΩôÁîüÂëΩÂÄºÔºö%düíñ\nÁßØÁ¥ØË¥¢ÂØåÔºö%düíé\nÊ∏∏ÊàèÊó∂ÈïøÔºö%d‰∏™Á≥ªÁªüÁßí", state[4], state[3], state[2], state[5]));
    }
    else{
        back_img.load(":/setting/image/bg_lose.png");
        ui->label1->setText(QString::asprintf("ÈùûÂ∏∏ÈÅóÊÜæÔºåËØ∑‰∏ãÊ¨°ÂÜçÊù•ÊåëÊàòÂêß„ÄÇ( ¬¥‚Ä¢Ã•Ã•Ã•œâ‚Ä¢Ã•Ã•Ã•` )"));
        ui->label2->setText(QString::asprintf("‰Ω†ÂÖ±ÂáªÊùÄ‰∫Ü%dÂè™ÊÄ™\nÁßØÁ¥ØË¥¢ÂØåÔºö%düíé\nÊ∏∏ÊàèÊó∂ÈïøÔºö%d‰∏™Á≥ªÁªüÁßí", state[4], state[2], state[5]));
    }
    QPixmap back_pixmap=QPixmap::fromImage(back_img);
    ui->label_2->setPixmap(back_pixmap);
}

// ËøôÊòØË∑≥ËΩ¨Âà∞ÊåáÂØºÁïåÈù¢
void ArkMidnight::on_btn_guide_clicked()
{
    guide1 *w = new guide1();
    int x = this->pos().x();
    int y = this->pos().y();
    w->move(x, y);
    w->show();
    this->setVisible(false);
    connect(w, &guide1::SignalInGuide_openArkMidnight, this, &ArkMidnight::vis);
}


// ËøôÊòØË∑≥ËΩ¨Âà∞Ê∏∏ÊàèÁïåÈù¢
void ArkMidnight::on_btn_start_clicked()
{
    ui->stackedWidget->setCurrentIndex(2);
}


void ArkMidnight::on_pushButton_clicked()
{

    game *w = new game(nullptr,
                       1,
                       {
    /*
    //0,1:normal
    //2,3:explode
    //4,5,6:fly
    //7,8:fast
    //Â∞ÜË¶ÅÂàõÈÄ†ÁöÑÊÄ™Áâ©ÂàóË°®ÔºåÊåâÂ±ÇÊ¨°‰æùÊ¨°‰∏∫Ôºö
    //1:Ê≥¢Ê¨°Ôºå2:Ê¨°Â∫èÔºå3:Â§öÂÖÉÊï∞ÁªÑ
    //Âú®3‰∏≠ 1:ÁßçÁ±ª, 2:Ë°ÄÈáèÔºå3:ÈÄüÂ∫¶ (1‰ª£Ë°®1s/1ÂÉèÁ¥†) 4:Êó∂Èó¥/timegap, ÂøÖÈ°ªÊåâÈ°∫Â∫èÊîæ
    */
                        { /* ÊÄ™Áâ©Âá∫Áé∞È°∫Â∫è */
                            {0,5,50,25},{1,5,50,125},{0,5,50,225},{1,5,50,325},{0,5,50,425},{1,5,50,525}
                        },
                        {
                            {0,25,50,100},{1,25,50,200},{0,25,50,300},{1,25,50,400},{7,15,100,500},{8,15,100,600}
                        },
                        {
                            {0,25,50,100},{1,25,50,200},{2,25,50,300},{3,25,50,400},{0,25,50,500},{1,25,50,600}
                        },
                        {
                            {4,15,50,100},{5,15,50,200},{6,15,50,300}
                        },
                        {
                            {0,25,50,100},{3,30,50,200},{5,20,50,200},{8,25,150,300}
                        },
                        }
                       ,
                       { /* ÂÖ≥Âç°Âú∞Âõæ */
                        {0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0},
                        {0, 1, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0},
                        {2, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0},
                        {0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 3},
                        {0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0},
                        {0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0},
                        {0, 0, 1, 0, 0, 0, 1, 0, 1, 0, 0, 0},
                        {0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0},
                        {0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0},
                        },
                       2000, /* ÈíªÁü≥ */
                       10, /* Ë°ÄÈáè */
                       {1, 2, 3, 4, 5, 6} /* ÂÖÅËÆ∏‰ΩøÁî®ÁöÑÁÇÆÂè∞ */
                       );

    int x = this->pos().x();
    int y = this->pos().y();
    w->move(x, y);
    w->show();
    this->setVisible(false);
    connect(w, &game::SignalInGame_openArkMidnight, this, &ArkMidnight::vis);
    connect(w, &game::SignalInGame_openArkMidnight2, this, &ArkMidnight::vis2);
}

void ArkMidnight::on_pushButton_2_clicked()
{
    //0,1:normal
    //2,3:explode
    //4,5,6:fly
    //7,8:fast
    //9,10:heal
    //11 heal,12 bomb,13 fly
    game *w = new game(nullptr,
                       2,
                       { /* ÊÄ™Áâ©Âá∫Áé∞È°∫Â∫è */
                        {
                            {0,15,50,100},{1,15,50,200},{0,15,50,300},{1,15,50,400},{0,25,50,500},{1,25,50,600}
                        },

                        {
                            {7,35,100,100},{8,35,100,200},{7,35,100,300},{8,35,100,400},{7,35,100,500},{8,35,100,600}
                        },
                        {
                            {0,75,40,100}, {1,75,40,200}, {2,75,40,300}, {7,45,100,350}, {3,100,40,400}, {8,55,100,450},
                            {0,100,40,500}, {7,55,100,550}, {1,114,40,600}, {8,65,100,650}, {1,114,40,700}, {8,65,100,750}
                        },
                        {
                            {0,150,40,100}, {4,20,50,150}, {1,175,40,200}, {4,25,50,250}, {0,150,40,300}, {4,25,50,350},
                            {1,200,40,400}, {6,30,50,450}, {1,200,40,500}, {6,30,50,550}, {2,125,40,600}, {6,10,50,650}
                        },
                        {
                            {0,350,40,100}, {4,50,50,150}, {1,350,40,200}, {4,50,50,250}, {2,400,40,300}, {4,60,50,350},
                            {3,400,40,400}, {5,60,50,450}, {0,500,40,500}, {5,60,50,550}, {1,500,40,600}, {6,100,50,650}
                        },
                        {
                            {1,450,40,100}, {6,90,50,150}, {1,450,40,200}, {6,90,50,250}, {9,200,40,300}, {6,90,50,350},
                            {10,200,40,400}, {6,90,50,450}, {1,550,40,500}, {6,100,50,550}, {1,550,40,600}, {6,100,50,650}
                        },
                        {
                            {9,1950,50,100},{10,1900,50,200},{9,1950,50,300},{10,1900,50,400},{9,1950,50,500},{10,1900,50,600},
                            {11,4001,50,700},
                            {9,1950,50,800},{10,1900,50,900},{9,1950,50,1000},{10,1900,50,1100},{9,1950,50,1200},{10,1900,50,1300}
                        },
                        {
                            {1,800,40,100}, {1,900,40,200}, {9,700,40,300},{10,700,40,400}, {1,900,40,500}, {1,1000,40,600},
                            {1,800,40,700}, {1,900,40,800}, {9,700,40,900},{10,700,40,1000}, {1,900,40,1100}, {1,1000,40,1200}
                        },
                        {
                            {1,1000,40,100},{7,500,100,150},{1,1000,40,200},{8,500,100,250},{9,1000,40,300},{7,500,100,350},
                            {10,1000,40,400},{8,500,100,450},{1,1000,40,500},{7,500,100,550},{1,1000,40,600},{8,500,100,650},
                            {2,1000,40,700},{7,500,100,750},{3,1000,40,800},{8,500,100,850},{9,1000,40,900},{7,500,100,950},
                            {10,1000,40,1000},{8,500,100,1050},{2,1000,40,1100},{7,500,100,1150},{3,1000,40,1200},{8,500,100,1250}
                        },
                        {
                            {4,250,40,100},{7,600,100,150},{6,300,40,200},{8,500,100,250},{4,250,40,300},{7,600,100,350},
                            {5,300,40,400},{8,500,100,450},{6,250,40,500},{7,600,100,550},{5,300,40,600},{8,500,100,650},
                            {4,250,40,700},{7,600,100,750},{6,300,40,800},{8,500,100,850},{4,250,40,900},{7,600,100,950},
                            {5,300,40,1000},{8,500,100,1050},{6,250,40,1100},{7,600,100,1150},{5,300,40,1200},{8,500,100,1250}
                        },
                        {
                            {12,2000,30,25},
                            {6,2,120,50},{6,2,120,100},{6,2,120,150},{6,2,120,200},{6,2,120,250},
                            {6,2,120,300},{6,2,120,350},{6,2,120,400},{6,2,120,450},{6,2,120,500},
                            {6,2,120,550},{6,2,120,600},{6,2,120,650},{6,2,120,700},{6,2,120,750},
                            {6,2,120,800},{6,2,120,850},{6,2,120,900},{6,2,120,950},{6,2,120,1000},
                            {6,2,120,1050},{6,2,120,1100},{6,2,120,1150},{6,2,120,1200},{6,2,120,1250},
                            {6,2,120,1300},{6,2,120,1350},{6,2,120,1400},{6,2,120,1450},{6,2,120,1500},
                            {6,2,120,1550},{6,2,120,1600},
                            {12,1000,30,1625},
                            {6,2,120,1650},{6,2,120,1700},{6,2,120,1750},{6,2,120,1800},{6,2,120,1850},
                            {6,2,120,1900},{6,2,120,1950},{6,2,120,2000},
                            {6,2,120,2050},{6,2,120,2100},{6,2,120,2150},{6,2,120,2200},{6,2,120,2250},
                            {6,2,120,2300},{6,2,120,2350},{6,2,120,2400},{6,2,120,2450},{6,2,120,2500},
                            {6,2,120,2550},{6,2,120,2600},{6,2,120,2650},{6,2,120,2700},{6,2,120,2750},
                            {6,2,120,2800},{6,2,120,2850},{6,2,120,2900},{6,2,120,2950},{6,2,120,3000},
                            {6,2,120,3050},{6,2,120,3100},{6,2,120,3150},{6,2,120,3200}
                        }
                        }
                       ,
                       { /* ÂÖ≥Âç°Âú∞Âõæ */
                        {2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
                        {0, 0, 0, 1, 0, 0, 0, 1, 1, 1, 0, 0},
                        {0, 0, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0},
                        {0, 1, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0},
                        {0, 1, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0},
                        {0, 1, 1, 1, 1, 1, 0, 0, 1, 0, 0, 0},
                        {0, 1, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0},
                        {0, 1, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0},
                        {0, 1, 0, 0, 0, 1, 0, 1, 1, 1, 0, 3},
                        },
                       1500, /* ÈíªÁü≥ */
                       5, /* Ë°ÄÈáè */
                       {1, 2, 3, 4, 5, 6} /* ÂÖÅËÆ∏‰ΩøÁî®ÁöÑÁÇÆÂè∞ */
                       );

    int x = this->pos().x();
    int y = this->pos().y();
    w->move(x, y);
    w->show();
    this->setVisible(false);
    connect(w, &game::SignalInGame_openArkMidnight, this, &ArkMidnight::vis);
    connect(w, &game::SignalInGame_openArkMidnight2, this, &ArkMidnight::vis2);
}

void ArkMidnight::on_pushButton_3_clicked()
{
    //0,1:normal
    //2,3:explode
    //4,5,6:fly
    //7,8:fast
    //9,10:heal
    //11 heal,12 bomb,13 fly
    game *w = new game(nullptr,
                       3,
                       { /* ÂÖ≥Âç°ÈúÄË¶ÅÊîπ */
                            {
                            {13,1500,10,50},
                            {6,2,120,100},{6,2,120,200},{6,2,120,300},{6,2,120,400},{6,2,120,500},
                            {6,2,120,650},{1,5,50,700},{6,2,120,750},{0,5,50,800},{6,2,120,850},{0,5,50,900},
                            {6,2,120,950},{1,10,50,1000},{6,2,120,1050},{0,10,50,1100},{6,2,120,1150},{0,10,50,1200},
                            {6,2,120,1250},{2,30,50,1300},{6,2,120,1350},{3,30,50,1400},{6,2,120,1450},{2,30,50,1500},
                            {6,2,120,1550},{2,35,50,1600},{6,2,120,1650},{3,42,50,1700},{6,2,120,1750},{2,35,50,1800},
                            {6,2,120,1850},{7,30,100,1900},{6,2,120,1950},{8,35,100,2000},{6,2,120,2050},{7,30,100,2100},
                            {6,2,120,2150},{7,32,100,2200},{6,2,120,2250},{8,37,100,2300},{6,2,120,2350},{7,32,100,2400},
                            {6,2,120,2450},{9,100,100,2500},{6,2,50,2550},{10,150,100,2600},{6,2,120,2650},{9,150,100,2700},
                            {6,2,120,2750},{9,106,100,2800},{6,2,56,2850},{10,156,100,2900},{6,2,120,2950},{9,156,100,3000}
                            }
                       }
                       ,
                       { /* ÂÖ≥Âç°ÈúÄË¶ÅÊîπ */
                        {9, 9, 0, 0, 0, 0, 0, 1, 1, 9, 9, 3},
                        {9, 0, 0, 1, 0, 0, 1, 0, 0, 0, 9, 0},
                        {0, 0, 1, 0, 0, 1, 0, 0, 9, 0, 1, 0},
                        {0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0},
                        {0, 1, 0, 1, 0, 0, 1, 0, 0, 1, 0, 0},
                        {0, 1, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0},
                        {0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0},
                        {0, 1, 0, 1, 0, 0, 1, 0, 0, 1, 0, 9},
                        {2, 1, 1, 0, 0, 0, 0, 0, 1, 0, 9, 9},
                        },
                       200,
                       1,
                       {1, 2, 3, 4, 5, 6}
                       );

    int x = this->pos().x();
    int y = this->pos().y();
    w->move(x, y);
    w->show();
    this->setVisible(false);
    connect(w, &game::SignalInGame_openArkMidnight, this, &ArkMidnight::vis);
    connect(w, &game::SignalInGame_openArkMidnight2, this, &ArkMidnight::vis2);
}

void ArkMidnight::on_pushButton_return_clicked()
{
    ui->stackedWidget->setCurrentIndex(0);
}


void ArkMidnight::on_pushButton_return_2_clicked()
{
    ui->stackedWidget->setCurrentIndex(2);
}

